crates/sui-source-validation-service/src/lib.rs

```diff
@@ -541,3 +513,48 @@ async fn api_route(
         )
     }
 }
+
+async fn check_version_header<B>(
+    headers: HeaderMap,
```

Comment:
It would be better to use `TypedHeader` for this, so that axum can help extract only the header you care about, rather than all of them.  The fullnode REST API has a good example of how this is done:

https://github.com/MystenLabs/sui/blob/12e127abb596a50ababa36cb46b2b49c2fb7a170/crates/sui-rest-api/src/checkpoints.rs#L34

The GraphQL version check has an apparently botched version of this, which I need to go fix, so don't use that as a reference.

Reply:
This is a good tip! After going down this path for this PR it started to become a rabbit hole -- defining a type and then implementing the header trait, finding the boiler, and then realizing the logic needs to change based on the current `Option` type, etc. etc., and I didn't exit the rabbit hole. So I'm going to punt for now and will follow up on this separately. Functionally we were doing the same thing as before, and the objective of this PR was just to factor out common stuff, so yes, I will revisit.